src/libsystemd/sd-journal/sd-journal.c

```diff
@@ -3154,7 +3154,7 @@ _public_ int sd_journal_enumerate_fields(sd_journal *j, const char **field) {
                         if (JOURNAL_HEADER_CONTAINS(of->header, n_fields) && le64toh(of->header->n_fields) <= 0)
                                 continue;
 
-                        r = journal_file_find_field_object_with_hash(of, o->field.payload, sz, le64toh(o->field.hash), NULL, NULL);
+                        r = journal_file_find_field_object(of, o->field.payload, sz, NULL, NULL);
```

Comment:
maybe do:

```c
if (!JOURNAL_HEADER_KEYED_HASH(f->header) && !JOURNAL_HEADER_KEYED_HASH(of->header))
        r = journal_file_find_field_object_with_hash(of, o->field.payload, sz, le64toh(o->field.hash), NULL, NULL);
else
        r = journal_file_find_field_object(of, o->field.payload, sz, NULL, NULL);
```

This retains the ability to compare across files if neither file uses the keyed hash logic.

Reply:
Will fix. I'm a bit surprised you say "retain" though, I was under the impression this change wouldn't change behavior at all. The proposed change seems solely a performance optimization to me so we don't needlessly calculate hashes if we don't need to. Is my understanding correct?

Reply:
yes.